tests/rptest/tests/maintenance_test.py

```diff
@@ -59,7 +59,12 @@ def _verify_broker_metadata(self, maintenance_enabled, node):
         mode is returning that draining has been enabled/disabled
         """
         node_id = self.redpanda.idx(node)
-        broker_target = self.admin.get_broker(node_id)
+        try:
```

Comment:
Can you say why this is the right place to handle the case of a broker not being visible on a node?

I'm unclear on how we end up in this situation.  The `registered()` check is supposed to wait until all the brokers are in visible to all the other brokers.  If it isn't doing that, we should fix the problem there.

Reply:
> Can you say why this is the right place to handle the case of a broker not being visible on a node?

It's a philosophical question. I think it's an ok place because it fixes an actual problem. But also... Do we expect our test to work in a safe environment or an environment closest to prod? I think the latter is more important so we when overload the test harness with safety checks we make the tests more sterile.

> I'm unclear on how we end up in this situation

Different code paths. `registered` uses `get_brokers` which in its turn uses `get_members_table`. `admin.get_broker` uses `get_broker` api which depends on `_metadata_cache`.

Reply:
it is really the same code, the `_metadata_cache` is a simple wrapper on members table, hmm i am wondering why this broker is not present in the result

Reply:
Folks, I think we should use less rigor when reviewing test fixes. It's an interesting question why broker api returns 404 but I don't think we should answer it in this PR. This PR is an already a n-th level follow up to some original PR and if we go deeper I bet something else pops up. I suggest to call it a day, open a ticket to investigate 404 but merge this fix and unblock the test.

Reply:
No, this kind of thing needs investigating.  When we change a test we should be able to say _why_ the fix is valid.

In this case, it's a redpanda bug.  `get_broker` in admin server is calling `get_node_drain_status` which can return an `errc::node_does_not_exist` error, which is turning into this 404.  The correct behavior there is to return a 200 with no drain status in the response (i.e. get_node_drain_status should probably just return a nullopt instead of an error).

Reply:
right, the problem was with the admin API returning 404 for an entity when its dependency fails, fixed it by returning 503 to avoid the REST misinterpretation